test/onnx/torchlib/test_ops.py

```diff
@@ -310,14 +316,52 @@ def test_output_match_opinfo_(
         self, device: str, dtype: torch.dtype, op: opinfo_core.OpInfo
     ):
         # Base test method for testing each op by running the full ONNX graph.
-        run_test_output_match(
-            self,
-            device,
-            dtype,
-            op,
-            ops_test_common.graph_executor,
-            ops_test_data.TORCHLIB_OPINFO_MAPPING,
-        )
+        opset_introduced = ops_test_data.TORCHLIB_OPINFO_MAPPING[
+            op.name
+        ].opset_introduced
+        if opset_introduced == 18:
+            run_test_output_match(
+                self,
+                device,
+                dtype,
+                op,
+                ops_test_common.graph_executor,
+                ops_test_data.TORCHLIB_OPINFO_MAPPING,
+            )
+
+    @ops_test_common.add_decorate_info(
+        ops_test_data.OPS_DB,
+        "TestOutputConsistencyFullGraph",
+        "test_output_match_opinfo_opset_",
+        skip_or_xfails=ops_test_data.EXPECTED_SKIPS_OR_FAILS,
+    )
+    @common_device_type.ops(  # type: ignore[misc]
+        [
+            info
+            for info in ops_test_data.OPS_DB
+            if info.name in ops_test_data.TESTED_OPS
+        ],
+        allowed_dtypes=TESTED_DTYPES,
+    )
+    def test_output_match_opinfo_all_opsets_(
+        self, device: str, dtype: torch.dtype, op: opinfo_core.OpInfo
+    ):
+        # Base test method for testing each op by running the full ONNX graph.
+        for opset_version in TESTED_OPSET_VERSIONS:
```

Comment:
Just making sure: we don’t loop over all upsets on all ops. Otherwise the test runtime grows linearly with the number of opsets we support, which can be large.

Reply:
Updated loop condition